### ISSUE TITLE
Query "too long to run" when no data found

### ISSUE THREAD

**Issue by markdlehane** [datagovuk/apiserver#80]
When querying the Transport data set (see SQL below) the response comments that the "Query would take too long to run, please apply a LIMIT of less than 5000 rows".  I suspect that there is no data available for the road, start_date and end_date and that the response should be changed to show that data is not available.

This is the query:

select  * 
from    planned_road_works 
where   road = 'M20' 
and     start_date > '01/01/2017' 
and     end_date < '07/01/2017'
limit   10

If I change the start and end dates to '01/01/2013' and '07/01/2013' respectively, data is returned.

Thank you.  This has fantastic potential.

**Comment by rossjones** [comment]
Yeah, we've had a few issues reported on this behaviour, I think there is data, but quite a lot of it with potentially some bad indexes :(

Unfortunately, as it's a proof of concept (as mentioned at https://data.gov.uk/data/api/) it's unlikely to be fixed before the service is withdrawn.  The aim of the data api was to obtain feedback about whether the service was something that people would like us to invest time in, but unfortunately we didn't get a lot.

**Comment by markdlehane** [comment]
Thanks for your reply. That's a shame. It seems very useful.
